Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js

```diff
@@ -1026,21 +1063,108 @@ class TimelineFromEndpoint {
+                    element.style.removeProperty("top");   
+                }
+            }
+        });
+        const placeHolderRef = REF.createRef({
+            state: {
+                show: false
             },
-            onSelectionScroll: (dots, selectedDotRect) => {
-                // this api will called with selected dots for each series once, and compose the selectedDotRect during the call
-                this.selectedDotsButtonGroupRef.setState({top: selectedDotRect.bottom, left: selectedDotRect.right});
+            onStateUpdate: (element, stateDiff, state) => {
+                const show = ("show" in stateDiff) ? stateDiff.show : state.show;
+                if (show)
+                    element.style.display = "block";
+                else
+                    element.style.display = "none";
+            }
+        });
+
+        const onScrollAction = (e) => {
+            const rect = containnerRef.element.getBoundingClientRect();
+            if (rect.top < 60 && 0 - rect.top < rect.height - 120) {
+                searchBarRef.setState({float: true});
+                placeHolderRef.setState({show: true});
+            } else {
+                searchBarRef.setState({float: false});
+                placeHolderRef.setState({show: false});
+            }
+        };
+        const onResizeAction = (e) => {
+            searchBarRef.element.style.width = `${containnerRef.element.getBoundingClientRect().width}px`;
+        };
+        const containnerRef = REF.createRef({
+            onElementMount: (element) => {
+                window.addEventListener("scroll", onScrollAction);
+                window.addEventListener("resize", onResizeAction);
+                onResizeAction();
             },
-        }, composer, ...children);
+            onElementUnmount: (element) => {
+                window.removeEventListener("scroll", onScrollAction);
+                window.addEventListener("resize", onResizeAction);
+            }
+        });
+        return `<div ref="${containnerRef}" style="position: relative">
+            <div ref="${searchBarRef}" class="next-regress-bar">
+                <button class="button" ref="${nextRegressButtonRef}">
+                    Next Regress Point
+                </button>
+            </div>
+            <div class="row" ref="${placeHolderRef}">
+                <div class="col-12" style="height: 40px; padding: 0"></div>
+            </div>
+            ${Timeline.CanvasContainer({
+                customizedLayer: `<div style="position:absolute" ref="${this.selectedDotsButtonGroupRef}"></div>`,
+                onSelecting: (e) => {
+                    this.selectedDotsButtonGroupRef.setState({show: false});
+                },
+                onSelect: (dots, selectedDotRect, seriesRect, e) => {
+                    // this api will called with selected dots for each series once, and compose the selectedDotRect during the call
```

Comment:
I don't know that this comment adds anything, especially when duplicated 3 lines down.